base/loading.jl

```diff
@@ -865,8 +865,26 @@ For more details regarding code loading, see the manual.
 function require(into::Module, mod::Symbol)
     uuidkey = identify_package(into, String(mod))
     # Core.println("require($(PkgId(into)), $mod) -> $uuidkey")
-    uuidkey === nothing &&
-        throw(ArgumentError("Module $mod not found in current path.\nRun `Pkg.add(\"$mod\")` to install the $mod package."))
+    if uuidkey === nothing
+        where = PkgId(into)
+        if where.uuid === nothing
+            throw(ArgumentError("""
+                Package $mod not found in current path:
```

Comment:
Suggesting just `Pkg.add` would perhaps make the message more light weight.

Reply:
Alternatively, you could say `not found in current LOAD_PATH`, to give the user a hint that such a thing exists and close the loop at bit for otherwise clueless users.